api/agent/drivers/docker/cookie.go

```diff
@@ -303,6 +353,17 @@ func (c *cookie) configureSecurity(log logrus.FieldLogger) {
 		"CapDrop": c.opts.HostConfig.CapDrop, "SecurityOpt": c.opts.HostConfig.SecurityOpt, "call_id": c.task.Id()}).Debug("setting security")
 }
 
+// addULimit adds lim to the docker host config ulimits slice, optionally
+// creating the slice if it isn't created yet.
+func (c *cookie) addULimit(lim docker.ULimit) {
+	limits := c.opts.HostConfig.Ulimits
+	if limits == nil {
+		limits = []docker.ULimit{}
+	}
```

Comment:
nit: this block (nil check) is unnecessary, appending to a nil slice is the same as appending to an empty, zero capacity, non-nil slice. I think this renders the `addULimit` method basically unnecessary, as each call site can simply be the one liner `c.opts.HostConfig.Ulimits = append(c.opts.HostConfig.Ulimits, limit)` rather than `addULimit`, but up to you - at least, this method can be a one liner.

Reply:
Good point, does look better that way :+1: